_posts/2019-12-20-ar-signpost.md

```diff
@@ -0,0 +1,254 @@
+
+<span class="image left"><img class="image fit" alt="AWS Fargate" src="/img/2019-12-20-ar-signpost/logo-nestjs.png" style="max-width: 100px;"></span>
+##### NestJS
+
+The framework used to create our backend is NestJS which is a framework based on NodeJS and TypeScript that abstracts you from the use of Express and Socket.io through decorators, has injection of dependencies "inspired" in Angular and allows to modularize our applications applying concepts of orientation to objects and functional and reactive programming.
+
+The official documentation is another of its strong points that you can find here and there is an official repository with many didactic examples.
+
+
+### Database
+
+We needed a database capable of representing real-world entities such as points of interest and their relationships in a graphical way and that's when we came across the fantastic database Neo4j.Below are the advantages offered by this graphical database.
+
+#### Used technologies
+
+<span class="image left"><img class="image fit" alt="AWS Fargate" src="/img/2019-12-20-ar-signpost/logo-neo4j.png" style="max-width: 100px;"></span>
+##### Neo4j
+
+Neo4j uses graphs to represent data and the relationships between them.
+A graph is defined as any graphical representation formed by vertices (illustrated by circles) and edges (shown by intersection lines).
+As shown in the image below.
+
+Databases oriented to graphs such as Neo4j perform better than relational (SQL) and non-relational (NoSQL).
+The key is that, although data queries increase exponentially, Neo4j's performance does not decrease, as opposed to relational DBs such as MySQL.
+
+## How it works
+
+At this point of blogpost, I will explain what this PoC project consists of and how it works in general terms.
+That said, let's get down to business!
+
+
+Now that these applications have been introduced, we are going to look at the interaction between them in order to better understand the functioning of the system as a whole.
+
+Let's start from the perspective of the user who has access to the mobile application as seen in the image below.
+
+<div style="text-align: center; margin: 0px auto;">
+ <a href="{{ '/img/2019-12-20-ar-signpost/ar-signpost-general-architecture.png' | prepend: site.baseurl }}" data-lightbox="ui" >
+<img alt="ikea space" src="{{ '/img/2019-12-20-ar-signpost/ar-signpost-general-architecture.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 750px;">
+</a>
+</div>
+
+
+Let's put your favorite mall on the scene for a moment.
+In this mall you would find markers with QR codes strategically placed so that they are visible, and you have easy access to them.
+Each QR contains contextual information such as a unique identifier, the building in which it is located and the floor in which it is located.
```

Comment:
Maybe: "Each QR contains contextual information such as a unique identifier and the building and floor in which it is located."